pkg/volume/flexvolume/flexvolume.go

```diff
@@ -93,22 +93,23 @@ func (plugin *flexVolumePlugin) GetAccessModes() []api.PersistentVolumeAccessMod
 }
 
 func (plugin *flexVolumePlugin) getVolumeSource(spec *volume.Spec) *api.FlexVolumeSource {
-	var source *api.FlexVolumeSource
-	if spec.Volume != nil && spec.Volume.FlexVolume != nil {
-		source = spec.Volume.FlexVolume
-	} else if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.FlexVolume != nil {
-		source = spec.PersistentVolume.Spec.FlexVolume
+	switch {
+	case spec.Volume != nil && spec.Volume.FlexVolume != nil:
```

Comment:
I changed it this way because I've seen this: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/plugins.go#L213